src/wristpy/core/orchestrator.py

```diff
@@ -159,14 +99,132 @@ def run(
+    will be taken from the given output path (if any). When the input path points to a
+    directory the output path must be a valid directory as well. Output file names will
+    be derived from original file names in the case of directory processing.
+
+
+    Args:
+        input: Path to the input file or directory of files to be read. Currently
+            supports .bin and .gt3x
+        output: Path to directory data will be saved to. If processing a single file the
+            path should end in the save file name in either .csv or .parquet formats.
+        thresholds: The cut points for the light, moderate, and vigorous thresholds,
+            given in that order. Values must be asscending, unique, and greater than 0.
+            Default values are optimized for subjects ages 7-11 [1].
+        calibrator: The calibrator to be used on the input data.
+        epoch_length: The temporal resolution in seconds, the data will be down sampled
+            to. If None is given no down sampling is preformed.
+        verbosity: The logging level for the logger.
+        output_filetype: Specifies the data format for the save files. Must be None when
+            processing files, must be a valid file type when processing directories.
+
+    Returns:
+        All calculated data in a save ready format as a Results object or as a
+        dictionary of Results objects.
+
+    Raises:
+        ValueError: If processing a file and the output_filetype is not None
+        ValueError: If processing a directory and output_filetype is not a valid type.
+        ValueError: If processing a directory and the output given is not a directory.
+        FileNotFoundError: If the input directory contained no files of a valid type.
+
+
+    References:
+        [1] Hildebrand, M., et al. (2014). Age group comparability of raw accelerometer
+        output from wrist- and hip-worn monitors. Medicine and Science in Sports and
+        Exercise, 46(9), 1816-1824.
+    """
+    logger.setLevel(verbosity)
+
+    input = pathlib.Path(input)
+
+    if input.is_file():
+        if output_filetype is not None:
+            raise ValueError(
+                "When processing single files, output_filetype should be None - "
+                "the file type will be determined from the output path."
+            )
+        logger.debug("Input is file, forwarding to run_file with output=%s", output)
+
+        return run_file(
+            input=input,
+            output=output,
+            thresholds=thresholds,
+            calibrator=calibrator,
+            epoch_length=epoch_length,
+            verbosity=verbosity,
+        )
+    if output is not None:
+        output = pathlib.Path(output)
+        if output.is_file():
+            raise ValueError(
+                "Output is a file, but must be a directory when input is a directory."
+            )
+
+    if output_filetype is None or output_filetype not in VALID_FILE_TYPES:
```

Comment:
Couldn't this just be
 
 `if output_filetype not in VALID_FILE_TYPES` since None is not in VALID_FILE_TYPES?

Reply:
This errors if output is None and output_filetype is None, but this should be fine right?

Reply:
Good point, I'll just tab it over once so that it's within the output is not None check

Reply:
To your first point I think i did that to appease a mypy error a little further down when I use with_suffix (not compatible with None). I'll just add another ignore there.